doc/Motr_LNet_Transport.rst

```diff
@@ -0,0 +1,623 @@
+
+- **Mapped memory page A memory page (struct page)**: that has been pinned in memory using the get_user_pages subroutine.
+
+- **Receive Network Buffer Pool**: This is a pool of network buffers, shared between several transfer machines. This common pool reduces the fragmentation of the cache of receive buffers in a network domain that would arise were each transfer machine to be individually provisioned with receive buffers. The actual staging and management of network buffers in the pool is provided through the [r.m0.net.network-buffer-pool] dependency.
+
+- **Transfer Machine Identifier**: This is an unsigned integer that is a component of the end point address of a transfer machine. The number identifies a unique instance of a transfer machine in the set of addresses that use the same 3-tuple of NID, PID and Portal Number. The transfer machine identifier is related to a portion of the Match bits address space in an LNet address - i.e. it is used in the ME associated with the receive queue of the transfer machine.
+
+Refer to [3], [5] and to net/net.h in the Motr source tree, for additional terms and definitions.
+
+***************
+Requirements
+***************
+
+- **[r.m0.net.rdma]** Remote DMA is supported. [2]
+
+- **[r.m0.net.ib]** Infiniband is supported. [2] 
+
+- **[r.m0.net.xprt.lnet.kernel]** Create an LNET transport in the kernel. [1] 
+
+- **[r.m0.net.xprt.lnet.user]** Create an LNET transport for user space. [1]
+
+- **[r.m0.net.xprt.lnet.user.multi-process]** Multiple user space processes can concurrently use the LNet transport. [1]
+
+- **[r.m0.net.xprt.lnet.user.no-gpl]** Do not get tainted with the use of GPL interfaces in the user space implementation. [1]
+
+- **[r.m0.net.xprt.lnet.user.min-syscalls]** Minimize the number of system calls required by the user space transport. [1]
+
+- **[r.m0.net.xprt.lnet.min-buffer-vm-setup]** Minimize the amount of virtual memory setup required for network buffers in the user space transport. [1]
+
+- **[r.m0.net.xprt.lnet.processor-affinity]** Provide optimizations based on processor affinity.
+
+- **[r.m0.net.buffer-event-delivery-control]** Provide control over the detection and delivery of network buffer events.
+
+- **[r.m0.net.xprt.lnet.buffer-registration]** Provide support for hardware optimization through buffer pre-registration.
+
+- **[r.m0.net.xprt.auto-provisioned-receive-buffer-pool]** Provide support for a pool of network buffers from which transfer machines can automatically be provisioned with receive buffers. Multiple transfer machines can share the same pool, but each transfer machine is only associated with a single pool. There can be multiple pools in a network domain, but a pool cannot span multiple network domains.
+
+******************
+Design Highlights
+******************
+
+The following figure shows the components of the proposed design and usage relationships between it and other related components:
+
+.. image:: Images/LNET.PNG
```

Comment:
These two image still have "c2_" prefix words.
Can you change it to "m0_" in the image?

Reply:
Sure. I will work on that.

Reply:
Incorporated.